doxia-core/src/main/java/org/apache/maven/doxia/sink/impl/AbstractLocator.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.maven.doxia.sink.impl;
+
+import org.apache.maven.doxia.sink.Locator;
+
+/**
+ * @since 2.0.0
+ */
+public abstract class AbstractLocator implements Locator {
+
+    private String reference;
+
+    protected AbstractLocator(String reference) {
+        super();
+        this.reference = reference;
+    }
+
+    @Override
+    public String getReference() {
+        return reference;
+    }
+
+    @Override
+    public String getLogPrefix() {
+        return formatLocation(this);
+    }
+
+    /**
+     * Creates a string with line/column information. Inspired by
+     * {@code o.a.m.model.building.ModelProblemUtils.formatLocation(...)}.
+     *
+     * @param locator The locator must not be {@code null}.
+     * @return The formatted location or an empty string if unknown, never {@code null}.
+     */
+    public static String formatLocation(Locator locator) {
```

Comment:
I believe that both `formatLocation` and `getLogPrefix` are misplaced here. This class is expected to contain on the four datapoints, formatting should be done else where, by the client cod, e.g. `AbstractParser` or so.

Reply:
This is used in `Sink` (not in `Parser`), therefore the only option would be to move to `AbstractSink`.

Reply:
Moved in https://github.com/apache/maven-doxia/pull/194/commits/ed298b000c4f95578755e8233cd0a4fb59dc2e8c.